### DISCUSSION TITLE
Can you use `init` in a script followed by other script commands?

### DISCUSSION THREAD

**Discussion by aytey** [x64dbg/x64dbg#2970] [Q&A]
## tl;dr

As title: can you use `init` in a script to load a binary, and then have the script carry on "as normal"?

## Longer version

I have a script that looks like this:

```
init C:\tmp\runme.exe
pause
bc
bphwc
bph 00007FF67F7210AE
run
```

That is, I am trying to script `x64dbg` to load `runme.exe`, delete all break points, and then run until `00007FF67F7210AE`.

However, if I comment-out the `pause`, then `x64gdb` gives me:

![image](https://user-images.githubusercontent.com/14277540/201487784-507a7c3d-9118-47cb-a6d2-2665ee372383.png)

(Line 3 is the `bc` because I've commented out the `pause`).

Note: if I load `runme.exe` "manually" (e.g., using File -> Open), and then have the script content be as above but without the `init` + `pause`, then the script does what I'd like.

Is what I'm trying to do possible?

## Version

![image](https://user-images.githubusercontent.com/14277540/201487911-0130aafc-12fd-465f-a519-4214383cdad7.png)

**Comment by mrexodia** [comment]
Currently this is not possible because of the way the commands work. The `init` command returns right away, before the system breakpoint.

Technically it’s likely possible to implement, but it would require some custom handling in the script engine. An alternative is to use a debuggee script and execute that when you start debugging something.
